src/app/view/components/dot-contentlet-editor/services/dot-add-contentlet.service.ts

```diff
@@ -0,0 +1,122 @@
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs/Subject';
+import { Observable } from 'rxjs/Observable';
+import { mergeMap, map, filter } from 'rxjs/operators';
+import { DotMenuService } from '../../../../api/services/dot-menu.service';
+
+interface DotAddEditEvents {
+    load?: ($event: any) => void;
+    keyDown?: ($event: any) => void;
+}
+
+export interface DotEditContentlet {
+    inode: string;
+    events?: DotAddEditEvents;
+}
+
+export interface DotAddContentLet {
+    type: string;
+    container: string;
+    events?: DotAddEditEvents;
+}
```

Comment:
I think maybe is better something like:

export interface DotEditContentlet {
    data: {
         inode: string;
    }
    events?: DotAddEditEvents;
}

export interface DotAddContentLet {
   data: {
       type: string;
       container: string;
   },
    events?: DotAddEditEvents;
}

Reply:
Why?

Reply:
Ok separate data from event